File: isegoria/src/main/java/com/eulersbridge/isegoria/auth/EmailVerificationFragment.java
package com.eulersbridge.isegoria.auth;

import android.arch.lifecycle.ViewModelProviders;
import android.os.Bundle;
import android.support.annotation.NonNull;
import android.support.v4.app.Fragment;
import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;
import android.widget.Button;

import com.eulersbridge.isegoria.R;

public class EmailVerificationFragment extends Fragment {

    private EmailVerificationViewModel viewModel;

    @Override
    public View onCreateView(@NonNull LayoutInflater inflater, ViewGroup container, Bundle savedInstanceState) {
        View rootView = inflater.inflate(R.layout.email_verification, container, false);

        viewModel = ViewModelProviders.of(this).get(EmailVerificationViewModel.class);

        //noinspection ConstantConditions
        AuthViewModel authViewModel = ViewModelProviders.of(getActivity()).get(AuthViewModel.class);

        final Button verifiedButton = rootView.findViewById(R.id.verifiedButton);
        verifiedButton.setOnClickListener(view -> {
            verifiedButton.setEnabled(false);

            authViewModel.verificationComplete.setValue(true);

            viewModel.userVerified().observe(this, success -> {
                if (success != null && !success)
                    verifiedButton.setEnabled(true);
            });
        });

        final Button resendVerificationButton = rootView.findViewById(R.id.resendVerificationButton);
        resendVerificationButton.setOnClickListener(view -> {
            view.setEnabled(false);

            viewModel.resendVerification().observe(this,
                    success -> view.setEnabled(true));
        });

        return rootView;
    }

    @Override
    public void onDestroy() {
        if (viewModel != null)
            viewModel.onExit();

        super.onDestroy();
    }
}

File: isegoria/src/main/java/com/eulersbridge/isegoria/auth/EmailVerificationFragment.kt
package com.eulersbridge.isegoria.auth

import android.arch.lifecycle.ViewModelProviders
import android.os.Bundle
import android.support.v4.app.Fragment
import android.view.LayoutInflater
import android.view.View
import android.view.ViewGroup
import com.eulersbridge.isegoria.R
import com.eulersbridge.isegoria.observe
import kotlinx.android.synthetic.main.email_verification.*

class EmailVerificationFragment : Fragment() {

    private val viewModel: EmailVerificationViewModel by lazy {
        ViewModelProviders.of(this).get(EmailVerificationViewModel::class.java)
    }

    override fun onCreateView(inflater: LayoutInflater, container: ViewGroup?, savedInstanceState: Bundle?): View?
        = inflater.inflate(R.layout.email_verification, container, false)

    override fun onViewCreated(view: View, savedInstanceState: Bundle?) {
        verifiedButton.setOnClickListener {
            it.isEnabled = false

            val authViewModel = ViewModelProviders.of(requireActivity()).get(AuthViewModel::class.java)
            authViewModel.verificationComplete.value = true

            observe(viewModel.userVerified()) {
                if (it == null || it == false)
                    verifiedButton.isEnabled = true
            }
        }

        resendVerificationButton.setOnClickListener { button ->
            button.isEnabled = false

            observe(viewModel.resendVerification()) {
                button.isEnabled = true
            }
        }
    }

    override fun onDestroy() {
        viewModel.onExit()

        super.onDestroy()
    }
}